src/main/java/io/cdap/e2e/pages/locators/CdfNameSpaceAdminLocators.java

```diff
@@ -0,0 +1,160 @@
+import org.openqa.selenium.support.How;
+
+/**
+ * Represents CdfNameSpaceAdminLocators
+ */
+public class CdfNameSpaceAdminLocators {
+
+  @FindBy(how = How.XPATH, using = "//div[contains(text(),'Namespace Admin')]")
+  public static WebElement namespaceAdminMenu;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='navbar-hamburger-icon']")
+  public static WebElement hamburgerMenu;
+
+  public static WebElement nameSpaceModules(String module, String nameSpace) {
+    String path = "//*[@href=\"/cdap/ns/" + nameSpace + "/details/" + module + "\"]";
+    return SeleniumDriver.getDriver().findElement(By.xpath(path));
+  }
+
+  public static WebElement createProfile(String nameSpace) {
+    String path = "//*[@href=\"/cdap/ns/" + nameSpace + "/profiles/create\"]";
+    return SeleniumDriver.getDriver().findElement(By.xpath(path));
+  }
+
+  public static WebElement switchNameSpace(String nameSpace) {
+    String path = "//*[@href=\"/cdap/ns/'" + nameSpace + "'\"]";
+    return SeleniumDriver.getDriver().findElement(By.xpath(path));
+  }
+
+  public static WebElement locateMenuLink(String menuLink) {
+    String xpath = "//*[@data-cy='navbar-" + menuLink + "-link']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateKeyProperty(String keyProperty, int row) {
+    String xpath = "//*[@data-cy='" + keyProperty + "" + row + "']//input[@placeholder='key']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateValueProperty(String valueProperty, int row) {
+    String xpath = "//*[@data-cy='" + valueProperty + "" + row + "']//input[@placeholder='value']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateAddRowButtonProperty(String addRowProperty, int row) {
+    String xpath =
+        "//*[@data-cy='" + addRowProperty + "" + row + "']//span"
+            + "//button[@type='submit' and @class='btn add-row-btn btn-link']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  @FindBy(how = How.XPATH, using = "//*[@class='namespace-dropdown']")
+  public static WebElement namespaceDropdown;
+
+  @FindBy(how = How.XPATH, using = "//div[contains(text(),'Add Namespace')]")
+  public static WebElement addNamespace;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-cy='wizard-next-btn']")
+  public static WebElement clickOnNextButton;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-cy='wizard-finish-btn']")
+  public static WebElement clickOnFinishButton;
+
+  public static WebElement locateKeyProperty(int row) {
+    String xpath = "//*[@data-cy= 'key-value-pair-" + row + "']//input[@placeholder='key']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateValueProperty(int row) {
+    String xpath = "//*[@data-cy= 'key-value-pair-" + row + "']//input[@placeholder='value']";
```

Comment:
prefer `data-testid` over `data-cy`.
Also locating elements based on the text of the placeholder value is not a good idea.